payu/experiment.py

```diff
@@ -387,7 +388,31 @@ def build_model(self):
         for model in self.models:
             model.build_model()
 
+    def check_payu_version(self):
+        """Check current payu version is greater than minimum required
+        payu version, if configured"""
+        # TODO: Move this function to a setup file if setup is moved to
+        # a separate file?
+        if "payu_minimum_version" not in self.config:
+            # Skip version check
+            return
+
+        minimum_version = str(self.config['payu_minimum_version'])
+
+        # Get the current version of the package
+        current_version = payu.__version__
+
+        # Compare versions
+        if version.parse(current_version) < version.parse(minimum_version):
```

Comment:
Maybe  do`version.parse(minimum_version)` when `minimum_version` is set above and wrap in a `try/except` block and tell the user they've stuffed up their version  specification before throwing an inscrutable python exception.

Reply:
Good point, I've moved the parsing earlier with hopefully a more helpful error message in a6bed42. I've added a couple tests with invalid versions.